### PR TITLE
Feature/autocomplete component

### PR DESCRIPTION
**Pull request by RohitKandimalla**
## MADiE PR

Jira Ticket: [MAT-4616](https://jira.cms.gov/browse/MAT-4616)
(Optional) Related Tickets:

### Summary

- Added a  new component AutoComplete which can be used for single-select and multi-select.
- Added a few stories to help us visualize, Used Template.bind for reusing a single template which reduces boiler plate code.
- Also made sure all 508 fixes are brought in, I hope I didn't miss any.

Few scenarios from storybook 

image
image
image
image
image

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

**Review by sb-cecilialiu — APPROVED** [review]

File: react/components/AutoComplete/AutoComplete.stories.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import React, { useState } from "react";
+import AutoComplete from "./AutoComplete";
+
+export default {
+    title: "AutoComplete",
+    component: AutoComplete,
+};
+
+const options = ["Option1", "Option2", "Option3", "Option4", "Option5"];
```
Comment (mcmcphillips) [inline react/components/AutoComplete/AutoComplete.stories.js:9]: Do we find it likely that this will be the shape of our options? 
My concern is our options may be objects that are required to send to the backend. This may be lost here.

Comment (mcmcphillips) [reply react/components/AutoComplete/AutoComplete.stories.js:9]: ah nevermind
I see AA can take values like 
const options = [
  { label: 'The Godfather', id: 1 },
  { label: 'Pulp Fiction', id: 2 },
];

**Review by mcmcphillips — APPROVED** [review]
